include/LocklessList.h

```diff
@@ -53,39 +53,29 @@ class LocklessList
 	{
 		Element * e = m_allocator->alloc();
 		e->value = value;
+		e->next = m_first.load(std::memory_order_relaxed);
 
-		do
+		while (!m_first.compare_exchange_weak(e->next, e,
+				std::memory_order_release,
+				std::memory_order_relaxed))
 		{
-#if QT_VERSION >= 0x050000
-			e->next = m_first.loadAcquire();
-#else
-			e->next = m_first;
-#endif
+			// Empty loop (compare_exchange_weak updates e->next)
```

Comment:
Why did you change this loop logic?

Reply:
I didn't; as mentioned in one of the commit messages, and in the comment here, `compare_exchange_*` functions update their first argument to the most recent value if they fail.